packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -176,25 +176,94 @@ function extractProjectsConfig(config: CliConfig): JsonObject {
       const outDir = app.outDir || defaults.outDir;
       const appRoot = app.root || defaults.appRoot;
 
-      const project: JsonObject = {
-        root: '',
-        projectType: 'application',
-        cli: {},
-        schematics: {},
-      };
+      function _mapAssets(asset: string | JsonObject) {
+        if (typeof asset === 'string') {
+          return { glob: asset, input: appRoot + '/', output: '/' + outDir + '/' };
+        } else {
+          return {
+            glob: asset.glob,
+            input: appRoot + '/' + asset.input,
+            output: outDir + '/' + asset.output,
+          };
+        }
+      }
+
+      function _buildConfigurations(): JsonObject {
+        const source = app.environmentSource;
+        const environments = app.environments;
+
+        if (!environments) {
+          return {};
+        }
+
+        return Object.keys(environments).reduce((acc, environment) => {
+          let isProduction = false;
+
+          const environmentContent = tree.read(app.root + '/' + environments[environment]);
+          if (environmentContent) {
+            isProduction = !!environmentContent.toString('utf-8')
+              // Allow for `production: true` or `production = true`. Best we can do to guess.
+              .match(/production['"]?\s*[:=]\s*true/);
+          }
+
+          acc[environment] = {
+            ...(isProduction
+              ? {
+                optimization: true,
+                outputHashing: 'all',
+                sourceMap: false,
+                extractCss: true,
+                namedChunks: false,
+                aot: true,
+                extractLicenses: true,
+                vendorChunk: false,
+                buildOptimizer: true,
+              }
+              : {}
+            ),
+            fileReplacements: [
+              { from: `${app.root}/${source}`, to: `${outDir}/${environments[environment]}` },
+            ],
+          };
+
+          return acc;
+        }, {} as JsonObject);
+      }
+
+      function _serveConfigurations(): JsonObject {
+        const environments = app.environments;
+
+        if (!environments) {
+          return {};
+        }
+
+        return Object.keys(environments).reduce((acc, environment) => {
+          acc[environment] = { browserTarget: `${name}:build:${environment}` };
 
-      const extraEntryMapper = (extraEntry: string | JsonObject) => {
+          return acc;
+        }, {} as JsonObject);
+      }
+
+      function _extraEntryMapper(extraEntry: string | JsonObject) {
         let entry: JsonObject;
         if (typeof extraEntry === 'string') {
           entry = { input: extraEntry };
         } else {
-          entry = extraEntry;
+          entry = {
+            input: extraEntry.input,
+            bundleName: extraEntry.output,
```

Comment:
You have to specialcase `bundleName` for extra entries that have `lazy:true`. They didn't need to specify the bundle name before because we calculated it via `basename` plus stripping the extension. But now we're doing it verbose so it needs the bundle name.

Reply:
You also need to pass on the `lazy` property, that isn't being passed on right now.

Reply:
Replaced with

```typescript
          const input = extraEntry.input as string;
          const bundleName =
            extraEntry.output
            || basename(
              normalize(input.replace(/\.(js|css|scss|sass|less|styl)$/i, '')),
            );
          entry = { input, bundleName };
```